### PR TITLE
Option to use system provided httplib

### PR DESCRIPTION
**Pull request by mattiaverga**
Some Linux distributions have their own httplib packaged, this will add an option to use the system provided one (default to use bundled library).

### CODE REVIEW

**Comment by knro** [comment]
System provided httplib requires CMake find file as well?

**Comment by mattiaverga** [comment]
> System provided httplib requires CMake find file as well?

Oh, yes, I haven't realized httplib provides CMake find file. 
Should I add a `find_package()` when the option is ON?

**Comment by knro** [comment]
Exactly

**Comment by mattiaverga** [comment]
Ok, not much cmake/C knowledge here, but I hope to have done it right...

File: CMakeLists.txt [hunk 1]
```diff
@@ -165,6 +168,20 @@ install(FILES
...
+# ##################################################################################################
+# ####################################  bundled libs  ##############################################
+# ##################################################################################################
+if(INDI_SYSTEM_HTTPLIB)
+    find_package(httplib REQUIRED)
+    include_directories(${HTTPLIB_INCLUDE_DIR})
+    set(SYSTEM_HTTPLIB 1)
+    message(STATUS "Using system provided httplib version ${HTTPLIB_VERSION}")
+else()
+    include_directories(libs/httplib)
```
Comment (knro) [inline CMakeLists.txt:180]: Maybe add ${CMAKE_CURRENT_SOURCE_DIR}/ prefix before libs?

**Comment by knro** [comment]
@pawel-soja Do you know what is causing the failure? I can reproduce the issue locally, it's related to the %include "indiapi.h" is this a new issue though?

**Comment by pawel-soja** [comment]
@knro I quickly looked through what was wrong and I think I found it. I've linked a new pull request to this thread. Once it is built, the problem is found and solved.

**Pull request merged**